libraries/AP_RangeFinder/AP_RangeFinder_MSP.cpp

```diff
@@ -59,7 +59,8 @@ void AP_RangeFinder_MSP::update(void)
 {
     //Time out on incoming data; if we don't get new
     //data in 500ms, dump it
-    if (AP_HAL::millis() - state.last_reading_ms > AP_RANGEFINDER_MSP_TIMEOUT_MS) {
+    const uint32_t last_update_ms = state.last_reading_ms;
+    if (AP_HAL::millis() - last_update_ms >= AP_RANGEFINDER_MSP_TIMEOUT_MS) {
```

Comment:
Not really sure why you're making the temporary variable, here and in other places.

Reply:
I think the issue is that we need to be sure that the last_reading_ms is older than the result of millis().  If we put it on the same line there's a small chance that it could be updated by another thread after we call millis().

Reply:
Mmm... I thought everything was on the same thread here.....  if not this is a good change.

If we *are*concerned about thread races here then you'd have to be concerned about `handle_msp` updating the timestamp, the main thread grabbing the data based on that timestamp and *then* `handle_msp` updating the distance...
 if you're really unlucky that data might be bad rather than stale...